variables.tf

```diff
@@ -0,0 +1,290 @@
+  type        = string
+  default     = "000"
+}
+
+variable "logical_product_family" {
+  type        = string
+  description = <<EOF
+    (Required) Name of the product family for which the resource is created.
+    Example: org_name, department_name.
+  EOF
+  nullable    = false
+
+  validation {
+    condition     = can(regex("^[_\\-A-Za-z0-9]+$", var.logical_product_family))
+    error_message = "The variable must contain letters, numbers, -, _, and .."
+  }
+}
+
+variable "logical_product_service" {
+  type        = string
+  description = <<EOF
+    (Required) Name of the product service for which the resource is created.
+    For example, backend, frontend, middleware etc.
+  EOF
+  nullable    = false
+
+  validation {
+    condition     = can(regex("^[_\\-A-Za-z0-9]+$", var.logical_product_service))
+    error_message = "The variable must contain letters, numbers, -, _, and .."
+  }
+}
+
+variable "use_azure_region_abbr" {
+  type        = bool
+  description = "Use Azure region abbreviation in the resource name"
+  default     = true
+}
+
+#########################################
+# Variables related to Role assignment
+#########################################
+
+variable "role_assignments" {
+  description = "A map of role assignments to be created"
+  type = map(object({
+    name                 = optional(string)
+    scope                = optional(string)
+    role_definition_name = string
+    principal_id         = optional(string)
+  }))
+  default = {}
+}
+
+###########################################
+# Variables related to private DNS zone
+###########################################
+
+variable "zone_name" {
+  type        = string
+  description = "Name of the private dns zone"
+  default     = "privatelink.vaultcore.azure.net"
+}
+
+variable "soa_record" {
+  type = object({
+    email        = string
+    expire_time  = number
+    minimum_ttl  = number
+    refresh_time = number
+    retry_time   = number
+    ttl          = number
+    tags         = map(string)
+  })
+  default = null
+}
+
+variable "private_dns_zone_tags" {
+  description = "Map of tags to be associated with the resource"
+  type        = map(string)
+  default     = {}
+}
+
+################################################
+# Variables related to private DNS zone link
+################################################
+
+variable "virtual_network_id" {
+  description = "The ID of the Virtual Network that should be linked to the DNS Zone. Changing this forces a new resource to be created."
+  type        = string
+}
+
+variable "registration_enabled" {
+  description = "(Optional) Is auto-registration of virtual machine records in the virtual network in the Private DNS zone enabled? Defaults to false."
+  type        = bool
+  default     = false
+}
+
+variable "private_dns_zone_link_vnet_tags" {
+  description = "Tags to be associated with this resource"
+  type        = map(string)
+  default     = {}
+}
+
```

Comment:
We should only have one set of tags named `tags` those should be passed in to all the modules that are wrapped in this ref arch. Since all these resources which are part of this module will have the same lifecycle I dont see a reason why we should have different tags for each module.